pkg/cache/cache.go

```diff
@@ -552,8 +558,9 @@ func (c *Cache) AddClusterQueue(ctx context.Context, cq *kueue.ClusterQueue) err
 			continue
 		}
 		c.addOrUpdateWorkload(&workloads.Items[i])
```

Comment:
doesn't this already handle the localQueue counters? Can we just remove the lines below?

Reply:
IIUC, this function doesn't update `admittedWorkloadsPerQueue`, right?
Do you mean, we don't need to update `admittedWorkloadsPerQueue `?

Reply:
I think it does, indirectly.

Reply:
That might be why the unit tests were passing before this PR

Reply:
Ah. I see.

Reply:
done.

Reply:
then... do we still need the set or can we just live with the integer?

Reply:
Uhm... If we use the set, we can avoid the potential issues. Although, we may be able to keep using integer for now.
We can come back to this discussion after we face this issue again.

Reply:
This means I will revert it.

Reply:
done.

Reply:
Thanks. I prefer we avoid an extra set for performance reasons.